src/platform/user/profile/vet360/containers/AddressValidationModal.jsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { get, map } from 'lodash';
+import Modal from '@department-of-veterans-affairs/formation-react/Modal';
+import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
+import { selectCurrentlyOpenEditModal } from '../selectors';
+import { openModal, createTransaction } from '../actions';
+
+import * as VET360 from '../constants';
+
+class AddressValidationModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedOption: '',
+      address: {},
+    };
+  }
+  onChangeHandler = address => event => {
+    this.setState({ selectedOption: event.target.name, address });
+  };
+
+  onSubmit = () => {
+    const { validationKey } = this.props;
+    const { address } = this.state;
+    const payload = { ...address, validationKey };
+
+    const method = payload.id ? 'PUT' : 'POST';
+
+    this.props.createTransaction(
+      VET360.API_ROUTES.EMAILS,
+      method,
+      this.props.fieldName,
+      payload,
+      this.props.analyticsSectionName,
+    );
+  };
+
+  renderWarningText = () => {
+    const { addressValidationError } = this.props;
+
+    return addressValidationError
+      ? "We're sorry. It looks like the address you entered isn't valid.  Please edit your address"
```

Comment:
convert the single straight quotes in the string to curly quotes (’) so they look better in the browser.

Reply:
Correct me if I'm wrong but I think the shortcut is - option + shift + ]

Reply:
It makes me feel like a skilled typist